### DISCUSSION TITLE
Means "Rule is stable" that it is enabled by default?

### DISCUSSION THREAD

**Discussion by buhtz** [astral-sh/ruff#10750] [Q&A]
I am looking into this list of rules and interpreting the icons behind each rule.

https://docs.astral.sh/ruff/rules/#legend

The green tick mark ("Rule is stable"), does it mean that the rule is active by default?

If not it would be nice to somehow see in the docs if a rule is contained in ruffs default behavior or not.

**Comment by MichaReiser** [comment]
> If not it would be nice to somehow see in the docs if a rule is contained in ruffs default behavior or not.

The default rules are `["E4", "E7", "E9", "F"]` as documented in the [settings page](https://docs.astral.sh/ruff/settings/#lint_select)

> The green tick mark ("Rule is stable"), does it mean that the rule is active by default?

No, it means it's not a preview rule, which is explained in the documentations:

> 🧪     The rule is unstable and is in ["preview"](https://docs.astral.sh/ruff/faq/#what-is-preview).

(Agree, it requires knowing that unstable is the opposite of stable)
